### PR TITLE
der: Custom error types in derive macros.

### PR DESCRIPTION
**Pull request by turbocool3r**
Add support for specifying a custom decode error type for `Sequence`, `Enumerated` and `Choice` derive macros using the `#[asn1(error = Ty)]` attribute.

This addresses #1559.

### CODE REVIEW

**Review by baloo — APPROVED** [review]
That looks good, the only thing that comes to mind would be to use a type to hold the attribute instead.

I've made a PR of what I had in mind in https://github.com/turbocool3r/formats/pull/1.

But it's mostly up to you.

**Comment by turbocool3r** [comment]
Thanks, just merged, that's a nice way to reduce that code duplication.

One last question is should the attribute be named something like `decode_error_type` to make its purpose more clear?

**Comment by baloo** [comment]
It's already `asn1(error=...)`, I think it's good enough.

File: der_derive/src/enumerated.rs [hunk 1]
```diff
@@ -24,6 +23,9 @@ pub(crate) struct DeriveEnumerated {
 
     /// Variants of this enum.
     variants: Vec<EnumeratedVariant>,
+
+    /// Error type for `DecodeValue` implementation.
+    error: Option<Path>,
```
Comment (baloo) [inline der_derive/src/enumerated.rs:28]: Did I forget this one?

Comment (baloo) [reply der_derive/src/enumerated.rs:28]: https://github.com/turbocool3r/formats/pull/2

**Comment by baloo** [comment]
I can merge that later tonight and make an rc.1 release for it.

**Comment by turbocool3r** [comment]
Merged the second PR, also added docs. I think it's ready for merge.

**Pull request merged**